components/ConversationsList.tsx

```diff
@@ -1,55 +1,51 @@
-import React, { useEffect } from 'react'
+import React from 'react'
 import Link from 'next/link'
 import { ChatIcon } from '@heroicons/react/outline'
 import Address from './Address'
 import { useRouter } from 'next/router'
 import { Conversation } from '@xmtp/xmtp-js'
-import { DecodedMessage } from '@xmtp/xmtp-js'
-import {
-  classNames,
-  truncate,
-  formatDate,
-  checkPath,
-  checkIfPathIsEns,
-} from '../helpers'
+import { classNames, formatDate, getConversationKey } from '../helpers'
 import Avatar from './Avatar'
 import { useAppStore } from '../store/app'
-import useWalletProvider from '../hooks/useWalletProvider'
 
 type ConversationTileProps = {
   conversation: Conversation
-  isSelected: boolean
   onClick?: () => void
 }
 
-const getLatestMessage = (messages: DecodedMessage[]): DecodedMessage | null =>
-  messages?.length ? messages[messages.length - 1] : null
-
 const ConversationTile = ({
   conversation,
-  isSelected,
   onClick,
 }: ConversationTileProps): JSX.Element | null => {
-  const convoMessages = useAppStore((state) => state.convoMessages)
+  const router = useRouter()
+  const address = useAppStore((state) => state.address)
+  const previewMessages = useAppStore((state) => state.previewMessages)
   const loadingConversations = useAppStore(
     (state) => state.loadingConversations
   )
+  const recipentAddress = Array.isArray(router.query.recipientWalletAddr)
+    ? router.query.recipientWalletAddr.join('/')
+    : router.query.recipientWalletAddr
 
-  if (!convoMessages.get(conversation.peerAddress)?.length) {
+  if (!previewMessages.get(getConversationKey(conversation))) {
     return null
   }
 
-  const latestMessage = getLatestMessage(
-    convoMessages.get(conversation.peerAddress) ?? []
-  )
-  const path = `/dm/${conversation.peerAddress}`
+  const latestMessage = previewMessages.get(getConversationKey(conversation))
+
+  const path = `/dm/${getConversationKey(conversation)}`
+
+  const conversationDomain =
+    conversation.context?.conversationId.split('.')[0] ?? ''
```

Comment:
While it's pretty to just display `Lens` instead of `lens.dev`, I don't know how workable that is as a general solution.

The only guidance we give to developers is that the conversationId should start with a domain. There may be multiple dots (`support.xmtp.org` and `support.relay.cc` would both show as `support` here), and there may be multiple apps with different TLDs (`xmtp.com` vs `xmtp.org` vs `xmtp.xyz`). All of that would get lost with this treatment.

Reply:
I would suggest splitting on the first `/`, even if it's less pretty for the Lens case.

We could always include a mapping of common apps to pretty names if we really wanted to show something nicer than `lens.xyz`